android/source/src/main/kotlin/com/gu/source/components/banner/AlertBanner.kt

```diff
@@ -0,0 +1,315 @@
+            vertical = 16.dp,
+            horizontal = 20.dp,
+        )
+
+        val iconAlertRound = Source.Icons.Base.AlertRound
+        val iconInfoRound = Source.Icons.Base.InfoRound
+        val iconCross = Source.Icons.Base.Cross
+
+        val textStyle = Source.Typography.TextSans17
+
+        val iconTextSpacing = 8.dp
+        val crossTextSpacing = 20.dp
+    }
+
+    /**
+     * Enum for the message type of the [AlertBanner]. Informs users of how important an action is.
+     *
+     * The messages are named based on banner message type and backdrop colour.
+     */
+    @Suppress("UndocumentedPublicProperty", "StringLiteralDuplication")
+    enum class Priority {
+        Neutral,
+        Informative,
+        Error,
+        ;
+
+        internal fun getBackdropColour() = when (this) {
+            Neutral -> AppColour(
+                light = Source.Palette.Neutral7,
+                dark = Source.Palette.Neutral20,
+            )
+
+            Informative -> AppColour(
+                light = Source.Palette.Brand800,
+                dark = Source.Palette.Brand400,
+            )
+
+            Error -> AppColour(
+                light = Source.Palette.News600,
+                dark = Source.Palette.News100,
+            )
+        }
+
+        internal fun getIcon(): ImageVector? = when (this) {
+            Informative -> Style.iconInfoRound
+
+            Error -> Style.iconAlertRound
+
+            Neutral -> null
+        }
+
+        internal fun getIconTint() = when (this) {
+            Informative -> AppColour(
+                light = Source.Palette.Brand400,
+                dark = Source.Palette.Brand800,
+            )
+
+            Error -> AppColour(
+                light = Source.Palette.Error400,
+                dark = Source.Palette.Error500,
+            )
+
+            Neutral -> AppColour.Unspecified
+        }
+
+        internal fun getTextColor() = when (this) {
+            Neutral -> AppColour(
+                light = Source.Palette.Neutral100,
+                dark = Source.Palette.Neutral100,
+            )
+
+            Informative -> AppColour(
+                light = Source.Palette.Neutral0,
+                dark = Source.Palette.Neutral100,
+            )
+
+            Error -> AppColour(
+                light = Source.Palette.Neutral0,
+                dark = Source.Palette.Neutral100,
+            )
+        }
+
+        internal fun getCrossIconTint() = when (this) {
+            Neutral -> AppColour(
+                light = Source.Palette.Neutral100,
+                dark = Source.Palette.Neutral100,
+            )
+
+            Informative -> AppColour(
+                light = Source.Palette.Neutral0,
+                dark = Source.Palette.Neutral100,
+            )
+
+            Error -> AppColour(
+                light = Source.Palette.Neutral0,
+                dark = Source.Palette.Neutral100,
+            )
+        }
+    }
+}
```

Comment:
[Required]: An Enum can have properties with limited visibility. You're hardcoding those properties as functions which requires evaluating that `when` multiple times. Instead, just make them properties of the enum:

```suggestion
    @Suppress("UndocumentedPublicProperty", "StringLiteralDuplication")
    /**
     * Enum for the message type of the [AlertBanner]. Informs users of how important an action is.
     *
     * The messages are named based on banner message type and backdrop colour.
     */
    @Suppress("UndocumentedPublicProperty", "StringLiteralDuplication")
    enum class Priority(
        internal val backgroundColour: AppColour,
        internal val iconTint: AppColour,
        internal val contentColour: AppColour,
        internal val icon: ImageVector?,
    ) {
        Neutral(
            backgroundColour = AppColour(
                light = Source.Palette.Neutral7,
                dark = Source.Palette.Neutral20,
            ),
            iconTint = AppColour.Unspecified,
            contentColour = AppColour(
                light = Source.Palette.Neutral100,
                dark = Source.Palette.Neutral100,
            ),
            icon = null,
        ),
        Informative(
            backgroundColour = AppColour(
                light = Source.Palette.Brand800,
                dark = Source.Palette.Brand400,
            ),
            iconTint = AppColour(
                light = Source.Palette.Brand400,
                dark = Source.Palette.Brand800,
            ),
            contentColour = AppColour(
                light = Source.Palette.Neutral0,
                dark = Source.Palette.Neutral100,
            ),
            icon = Source.Icons.Base.InfoRound,
        ),
        Error(
            backgroundColour = AppColour(
                light = Source.Palette.News600,
                dark = Source.Palette.News100,
            ),
            iconTint = AppColour(
                light = Source.Palette.Error400,
                dark = Source.Palette.Error500,
            ),
            contentColour = AppColour(
                light = Source.Palette.Neutral0,
                dark = Source.Palette.Neutral100,
            ),
            icon = Source.Icons.Base.AlertRound,
        ),
        ;
    }
}
```